pkg/generate/app/componentref.go

```diff
@@ -44,10 +44,6 @@ func componentWithSource(s string) (component, repo string, builder bool, err er
 	} else {
 		component = s
 	}
-	// component must be of the form compatible with a pull spec *or* <namespace>/<name>
-	if _, err := imageapi.ParseDockerImageReference(component); err != nil {
```

Comment:
I am not even sure this gets called when you do `oc new-app -f ...`. That file argument goes straight to the template file bucket, there are no arguments to detect, this shouldn't be called.